app/ante/cosmos/min_price.go

```diff
@@ -0,0 +1,109 @@
+package cosmos
+
+import (
+	"fmt"
+	"math/big"
+
+	"golang.org/x/exp/slices"
+
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	ethante "github.com/evmos/ethermint/app/ante"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// MinGasPriceDecorator will check if the transaction's fee is at least as large
+// as the MinGasPrices param. If fee is too low, decorator returns error and tx
+// is rejected. This applies for both CheckTx and DeliverTx
+// If fee is high enough, then call next AnteHandler
+// CONTRACT: Tx must implement FeeTx to use MinGasPriceDecorator
+type MinGasPriceDecorator struct {
+	feeMarketKeeper ethante.FeeMarketKeeper
+	evmKeeper       ethante.EVMKeeper
+	feeKeeper       FeeKeeper
+}
+
+// NewMinGasPriceDecorator creates a new MinGasPriceDecorator instance used only for
+// Cosmos transactions.
+func NewMinGasPriceDecorator(
+	fmk ethante.FeeMarketKeeper,
+	ek ethante.EVMKeeper,
+	fk FeeKeeper,
+) MinGasPriceDecorator {
+	return MinGasPriceDecorator{
+		feeMarketKeeper: fmk,
+		evmKeeper:       ek,
+		feeKeeper:       fk,
+	}
+}
+
+func (mpd MinGasPriceDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+	// Check if the transaction contains any non-free messages
+	hasNonFreeMsg := mpd.feeKeeper.HasNonFeeTx(ctx, tx)
+	if !hasNonFreeMsg {
+		return next(ctx, tx, simulate)
+	}
```

Comment:
```suggestion
	hasNonFreeMsg := mpd.feeKeeper.HasNonFeeTx(ctx, tx)
	if !hasNonFreeMsg {
		return next(ctx, tx, simulate)
	}
```
only check cosmos transactions(msg)